File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] result = new int[nums.length];
        int left = 0;
        int right = nums.length - 1;
        for (int i = nums.length-1; i >= 0; i--) {
            int nLeft = nums[left] * nums[left];
            int nRight = nums[right] * nums[right];
            if (nLeft > nRight) {
                result[i] = nLeft;
                ++left;
            } else {
                result[i] = nRight;
                --right;
            }
        }
        return result;
    }
}

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0] * n
        left = 0
        right = n-1
        for i in range(n-1, -1, -1):
            cur = 0
            if nums[left]**2 > nums[right]**2:
                res[i] = nums[left]**2
                left += 1
            else:
                res[i] = nums[right]**2
                right -= 1
        return res